realsense2_camera/src/t265_realsense_node.cpp

```diff
@@ -144,3 +165,69 @@ void T265RealsenseNode::warningDiagnostic(diagnostic_updater::DiagnosticStatusWr
 {
   status.summary(diagnostic_msgs::DiagnosticStatus::WARN, _T265_fault);
 }
+
+bool T265RealsenseNode::importLocalizationMap(const std::string &localization_file)
+{
+    try
+    {
+      _pose_sensor.import_localization_map(this->bytesFromRawFile(localization_file));
+      ROS_INFO_STREAM("T265 Localization map loaded from " << localization_file);
+    }
+    catch (std::runtime_error& e)
+    {
+      ROS_WARN_STREAM("Error loading T265 map from " << localization_file << ": " << e.what());
+      return false;
+    }
+    return true;
+}
+
+bool T265RealsenseNode::exportLocalizationMap(const std::string &localization_file)
+{
+    try
+    {
+        this->rawFileFromBytes(localization_file, _pose_sensor.export_localization_map());
+        ROS_INFO_STREAM("Saved T265 map to " << localization_file);
+    }
+    catch (std::runtime_error& e)
+    {
+        ROS_WARN_STREAM("Error saving T265 map to " << localization_file << ": " << e.what());
+        return false;
+    }
+    return true;
+}
+
+bool T265RealsenseNode::saveRelocalizationMapSrv(realsense2_camera::MapPathString::Request &req,
+                                                 realsense2_camera::MapPathString::Response &res) {
+    res.success = this->exportLocalizationMap(req.filepath);
+    return true;
+}
+
+std::vector<uint8_t> T265RealsenseNode::bytesFromRawFile(const std::string &filename)
+{
+    std::ifstream file(filename.c_str(), std::ios::binary);
+    if (!file.good())
+        throw std::runtime_error("Invalid binary file specified. Verify the source path and location permissions");
+
+    // Determine the file length
+    file.seekg(0, std::ios_base::end);
+    std::size_t size = file.tellg();
+    if (!size)
+        throw std::runtime_error("Invalid binary file -zero-size");
+    file.seekg(0, std::ios_base::beg);
+
+    // Create a vector to store the data
+    std::vector<uint8_t> v(size);
+
+    // Load the data
+    file.read((char*)&v[0], size);
+
+    return v;
+}
+
+void T265RealsenseNode::rawFileFromBytes(const std::string &filename, const std::vector<uint8_t> &bytes)

```

Comment:
this is pretty much whatever was in `raw_file_from_bytes` within https://github.com/IntelRealSense/librealsense/blob/master/examples/ar-advanced/rs-ar-advanced.cpp